lacework/resource_lacework_integration_aws_agentless_scanning.go

```diff
@@ -0,0 +1,245 @@
+		Required:    true,
+		Description: "How often in hours the scan will run in hours.",
+	},
+	"scan_containers": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: "Whether to includes scanning for containers.",
+	},
+	"scan_host_vulnerabilities": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: "Whether to includes scanning for host vulnerabilities.",
+	},
+	"enabled": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     true,
+		Description: "The state of the external integration.",
+	},
+	"retries": {
+		Type:        schema.TypeInt,
+		Optional:    true,
+		Default:     5,
+		Description: "The number of attempts to create the external integration.",
+	},
+	"created_or_updated_time": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"created_or_updated_by": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"type_name": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"org_level": {
+		Type:     schema.TypeBool,
+		Computed: true,
+	},
+	"server_token": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"uri": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+}
+
+func resourceLaceworkIntegrationAwsAgentlessScanningCreate(d *schema.ResourceData, meta interface{}) error {
+	var (
+		lacework = meta.(*api.Client)
+		retries  = d.Get("retries").(int)
+	)
+
+	awsAgentlessScanningData := api.AwsSidekickData{
+		ScanFrequency:           d.Get("scan_frequency").(int),
+		ScanContainers:          d.Get("scan_containers").(bool),
+		ScanHostVulnerabilities: d.Get("scan_host_vulnerabilities").(bool),
+	}
+
+	if d.Get("query_text") != nil {
+		awsAgentlessScanningData.QueryText = d.Get("query_text").(string)
+	}
+
+	awsAgentlessScanning := api.NewCloudAccount(d.Get("name").(string),
+		api.AwsSidekickCloudAccount,
+		awsAgentlessScanningData,
+	)
+
+	if !d.Get("enabled").(bool) {
+		awsAgentlessScanning.Enabled = 0
+	}
+
+	return resource.RetryContext(context.Background(), d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		retries--
+		log.Printf("[INFO] Creating %s cloud account integration\n", api.AwsSidekickCloudAccount.String())
+		response, err := lacework.V2.CloudAccounts.CreateAwsSidekick(awsAgentlessScanning)
+		if err != nil {
+			if retries <= 0 {
+				return resource.NonRetryableError(
+					fmt.Errorf("Error creating %s cloud account integration: %s",
+						api.AwsSidekickCloudAccount.String(), err,
+					))
+			}
+			log.Printf(
+				"[INFO] Unable to create %s cloud account integration. (retrying %d more time(s))\n%s\n",
+				api.AwsSidekickCloudAccount.String(), retries, err,
+			)
+			return resource.RetryableError(fmt.Errorf(
+				"Unable to create %s cloud account integration (retrying %d more time(s))",
+				api.AwsSidekickCloudAccount.String(), retries,
+			))
+		}
+
+		cloudAccount := response.Data
+		d.SetId(cloudAccount.IntgGuid)
```

Comment:
Just noticing this is the same as eks audit log. Wondering if I made a mistake with that. Should it be `cloudAccount.ID()`?

Reply:
Actually.... NVM I see that `cloudAccount.ID()` actually just returns the `IntgGuid`